lib/util/middleware.js

```diff
@@ -0,0 +1,64 @@
+if (typeof (fetch) === 'undefined') require('isomorphic-fetch')
+
+const API_USER_PATH = '/api/secure/user'
+
+export async function secureFetch (url, accessToken, apiKey, method = 'get', options = {}) {
+  const res = await fetch(url, {
+    method,
+    mode: 'cors', // Middleware is at a different URL.
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      'Content-Type': 'application/json',
+      'x-api-key': apiKey
+    },
+    ...options
+  })
+  if ((res.status && res.status >= 400) || (res.code && res.code >= 400)) {
+    const result = await res.json()
+    let message = `Error ${method}-ing user: ${result.message}`
+    if (result.detail) message += `  (${result.detail})`
+
+    console.log(message)
+    return {
+      status: 'error',
+      message
+    }
+  }
+  return {
+    status: 'success',
+    data: await res.json()
+  }
+}
+
+export async function fetchUser (middlewareConfig, idField, id, token) {
```

Comment:
Needs a comment that this method will fetch a user by their Auth0Id and not the Mongo Id.

Reply:
Might want to consider renaming to `fetchUserByAuth0Id`.

Reply:
The methods in here are agnostic to Auth0 (even though its caller specifies `idField` to be the Auth0 id).

Reply:
@binh-dam-ibigroup, it's very unlikely that this code will ever be refactored to not use Auth0. I appreciate your efforts to isolate Auth0 and make certain auth things generic, but we need the code in its present state to be readable when we revisit and actually reflect the components that its built with. In my view, part of that would be to make naming changes like @evansiroky suggested above.